docs/community-adapters/guide.md

```
@@ -0,0 +1,122 @@
+```
+5. **Define the model input modalities**: Every model usually supports different input modalities (like text, images, etc). After you define the model metadata, you need to implement the input modalities the model supports. Generally you want to
+do something like this:
+```typescript
+export type OpenAIModelInputModalitiesByName = {
+  [GPT5_2.name]: typeof GPT5_2.supports.input
+  [GPT5_2_PRO.name]: typeof GPT5_2_PRO.supports.input
+  [GPT5_2_CHAT.name]: typeof GPT5_2_CHAT.supports.input
+  //  ... repeat for each model
+}
+```
+6. **Define your model options**: After you define the model metadata, you need to implement the model options the model supports. Generally you want to do something like this (you can see an example for OpenAI models [here](https://github.com/TanStack/ai/blob/main/packages/typescript/ai-openai/src/text/text-provider-options.ts)):
```

markdownlint-cli2 (0.18.1)

Link text should be descriptive

(MD059, descriptive-link-text)